test/05-webresource.test.ts

```diff
@@ -0,0 +1,472 @@
+	const contentType = 'image/png';
+	let fileSize: number;
+
+	before(async () => {
+		pineServer = await testInit({
+			configPath,
+			hooksPath,
+			deleteDb: true,
+		});
+		const fileInfo = await fs.stat(filePath);
+		fileSize = fileInfo.size;
+	});
+
+	after(async () => {
+		testDeInit(pineServer);
+	});
+
+	const transalations = [
+		{ resourcePath: 'logo_image', resourceName: 'example' },
+		{ resourcePath: 'other_image', resourceName: 'v1' },
+		{ resourcePath: 'not_translated_webresource', resourceName: 'example' },
+		{ resourcePath: 'not_translated_webresource', resourceName: 'v1' },
+	];
+
+	transalations.forEach(({ resourcePath, resourceName }) => {
+		describe(`webresource ${resourcePath} - ${resourceName}`, () => {
+			it(`creates a organization with a ${resourcePath}`, async () => {
+				const { body: organization, headers } = await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, filePath, { filename, contentType })
+					.expect(201);
+
+				expect(organization[resourcePath].size).to.equals(fileSize);
+				expect(organization[resourcePath].filename).to.equals(filename);
+				expect(organization[resourcePath].contentType).to.equals(contentType);
+
+				const getRes = await supertest(testLocalServer)
+					.get(headers.location)
+					.expect(200);
+
+				expect(getRes.body.d[0]).to.deep.equal(organization);
+
+				const { body: photoRes } = await supertest(
+					organization[resourcePath].href,
+				)
+					.get('')
+					.set({
+						responseType: 'arraybuffer',
+						headers: {
+							Accept: '*/*',
+						},
+					})
+					.expect(200);
+
+				await expectToExist(organization[resourcePath].filename);
+				const receivedSize = photoRes.length;
+				expect(receivedSize).to.equals(fileSize);
+				const realImage = await fs.readFile(filePath);
+				const diff = realImage.compare(photoRes);
+				expect(diff).to.be.eq(0);
+			});
+
+			it(`does not store ${resourcePath} if is bigger than PINEJS_WEBRESOURCE_MAXFILESIZE`, async () => {
+				const uniqueFilename = `${randomUUID()}_${filename}`;
+				const { largeStream } = await getLargeFileStream(
+					intVar('PINEJS_WEBRESOURCE_MAXFILESIZE') + 10 * 1024 * 1024,
+					filePath,
+				);
+				const res = await supertest(testLocalServer)
+					.post(`/${resourceName}/organization`)
+					.field('name', 'John')
+					.attach(resourcePath, largeStream, {
+						filename: uniqueFilename,
+						contentType,
+					})
+					.expect(400);
+
+				expect(res.body).to.include('File size exceeded');
+
+				await expectNotToExist(uniqueFilename);
+			});
+
+			it(`creates a organization with a large ${resourcePath}`, async () => {
+				// Create a large file by using an image at the head ( so that filetype can identify it as an image ),
+				// and then appending a large chunk
+				const { largeStream, largeFileSize } = await getLargeFileStream(
+					1024 * 1024 * 512 - fileSize,
```

Comment:
Shouldn't this file size fail?

Reply:
This creates a filestream that is slightly smaller than the size on the SBVR.